### ISSUE TITLE
When I call getWeather I'm getting a null safety exception

### ISSUE THREAD

**Issue by sowens-csd** [tanzilzubair/open_weather_api_client#1]
When I call 
```dart
OneCallWeatherFactory weatherFactory = OneCallWeatherFactory(
      apiKey: "my-api-key",
      settings: UnitSettings(
        windSpeedUnit: WindSpeedUnit.Kilometres_Hour,
      ),
      locationCoords: LocationCoords(
        longitude: 51.5074,
        latitude: 0.1278,
      ),
    );
RequestResponse<OneCallWeather?> result = await weatherFactory.getWeather();
```

I'm getting this error:
```
_CastError (Null check operator used on a null value)
```

Coming from `one_call_weather_model.dart` line 56, which is:

```dart
 List<OneCallMinutelyWeather?>? minutelyWeather = minutelyPayload!.map(
```

Thoughts?

**Comment by tanzilzubair** [comment]
I’m a little busy right now so I haven’t been able to test this out, but what I believe is happening is that the location your calling the weather for did not have rainfall or snowfall during this particular hour. 

As such, the server returned nothing for the parameters OneCallMinutelyWeather is trying to access, which is why the null check fails, which should probably be fixed through some form of null checking or making the OneCallMinutelyWeather nullable. I’m leaning towards the latter, but your thoughts on the matter would also be appreciated greatly.

Regardless of whether this is the issue, the error seems to be entirely in my implementation, and I’ll get on fixing it ASAP. Thank you for choosing my package over all the others, and if you have any other questions, feedback, advice or feature requests, please do not hesitate to open an issue or comment here.

**Comment by tanzilzubair** [comment]
@sowens-csd I checked out the code, and from the testing I've done, it seems that the location you're trying to search for is in the middle of the Indian Ocean. 

I generated the url corresponding to the query you showed in your code, and upon querying it manually and looking at the server response, I found that OpenWeather doesn't seem to provide minutely forecasts for ocean locations. The current weather, the hourly weather, and the daily weather, were however, still provided.

The code, however, should not have failed, and I will be publishing an updated version of the package shortly that fixes this bug, and returns null if the server does not return any data for a particular item in the query.

Again, thank you for choosing my package over all the others, and if you have any other questions, feedback, advice or feature requests, please do not hesitate to open an issue or comment here.

**Comment by tanzilzubair** [comment]
@sowens-csd I have added the appropriate checks and published a new version of the package, V-2.0.1, that you can find [here](https://pub.dev/packages/open_weather_api_client) . As an update addressing the error has been published, and you have not responded, I am assuming your problem has been fixed and you have not faced further problems using this package, and I am closing this issue.

Once again, thank you for choosing my package over all the others, and if you have any further questions, feedback, advice or feature requests, or face any further problems, please do not hesitate to open an issue.

**Comment by sowens-csd** [comment]
Thanks for the quick feedback and the update. The coordinates in the post above were part of an edit, I was actually searching for Toronto. I think you've fixed this issue, on to another one. I'll post a new issue for that.
